fferr/errors.go

```diff
@@ -75,8 +76,8 @@ type Error interface {
 	GetType() string
 	GRPCStatus() *status.Status
 	ToErr() error
-	AddDetail(key, value string)
-	AddDetails(keysAndValues ...interface{})
+	AddDetail(key, value string) Error
```

Comment:
I just pulled your branch and figured out why I didn't do this. On Goland (and probably vscode) we all now get warnings that there is an unhandled error.  Returning the error could be confusing because user isn't sure if it's the error that's being returned or an error for adding the detail

Reply:
Its a pretty common pattern, what do you recommend?